pkg/api/v1beta3/types.go

```diff
@@ -200,26 +200,26 @@ type VolumeSource struct {
 	// to see the host machine. Most containers will NOT need this.
```

Comment:
It's ok to leave them all omitempty for now, but I can't think of a scenario where metadata would be absent. We could maybe extract the name from the URLs on some operations, but I don't think we do that currently.